src/container/container_iv.c

```diff
@@ -1275,7 +1275,10 @@ cont_iv_prop_g2l(struct cont_iv_prop *iv_prop, daos_prop_t **prop_out)
 		prop_entry = &prop->dpp_entries[i++];
 		acl = &iv_prop->cip_acl;
 		if (acl->dal_ver != 0) {
-			D_ASSERT(daos_acl_validate(acl) == 0);
+			rc = daos_acl_validate(acl);
+			if (rc == -DER_NOMEM)
+				D_GOTO(out, rc);
+			D_ASSERT(rc == 0);
```

Comment:
For some reason I thought daos_acl_validate() returned a boolean but I see now it's an int.

This function is validating input from the client, if the client sends a invalid acl then the server should not fail so I wonder if this patch could be even simpler.
```suggestion
			rc = daos_acl_validate(acl);
			if (rc != -DER_SUCCESS)
				D_GOTO(out, rc);
```